pkg/types/gcp/validation/platform.go

```diff
@@ -69,3 +71,12 @@ func ValidatePlatform(p *gcp.Platform, fldPath *field.Path) field.ErrorList {
 
 	return allErrs
 }
+
+// ValidateClusterName confirms that the provided cluster name matches GCP naming requirements.
+func ValidateClusterName(clusterName string) error {
+	re := regexp.MustCompile(`(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?)`)
+	if re.FindString(clusterName) != clusterName {
```

Comment:
matchstring vs find string ?
https://github.com/kubernetes/apimachinery/blob/adc6f4cd9e7d28fc1c1e5efc658d940d55c0f356/pkg/util/validation/validation.go#L149-L151

Reply:
When I use MatchString I get false positives using the regex here (which comes from the gcp api error message in the bz): 
https://play.golang.org/p/edq7rew99Qo

That is because match string just looks for a match of the regex in the string. I looked at the code you linked and it works, but I'm not sure how it enforces that the whole string match. I'm assuming it forces the the match through the regular expression itself, but I need to look into it a bit more. 

I'm not sure, but it doesn't seem like we could use MatchString without changing the regex. Any suggestions would be appreciated.

Reply:
https://github.com/kubernetes/apimachinery/blob/adc6f4cd9e7d28fc1c1e5efc658d940d55c0f356/pkg/util/validation/validation.go#L140

If you look at the regex, it is anchored with word begining and word end to enforce complete match.

Reply:
I see it now. Will follow this pattern. Thanks!